pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdSinks.java

```diff
@@ -469,20 +477,56 @@ protected FunctionDetails createSinkConfig(SinkConfig sinkConfig) throws IOExcep
             // set source spec
             // source spec classname should be empty so that the default pulsar source will be used
             SourceSpec.Builder sourceSpecBuilder = SourceSpec.newBuilder();
+            sourceSpecBuilder.setSubscriptionType(Function.SubscriptionType.SHARED);
```

Comment:
I think it is still very confused about

`--inputs`
`--topics-pattern`
`--customSerdeInputs`
`--customSchemaInputs`

@merlimat @srkukarni - can you guys describe what is the eventual options for users to use? any options should be marked as hidden (to deprecate)

Reply:
The pr makes it backwards compatible to whats existing out there. Just like in functions, the idea is that
1) --inputs is a comma separated list of topics that uses default schema/serde
2) --topics-pattern is topic pattern that uses default schema/serde
3) customSerdeInputs is a map from topicname -> custom serde classname

These above are whats available today already. The newly available one is 
4) customSchemaInputs is very similar to 3) but with schemas

Reply:
yes. but the existing options are very confused. what is the plan to address the confusion?

Reply:
I specifically don't have anything to address that currently. Any suggestions from you/others is welcome

Reply:
we should at least hide --customSerdeInputs since serdes are deprecated

Reply:
serde's are not deprecated. Only DefaultSerde is.